### PR TITLE
S-23 to master

### PR DESCRIPTION
**Pull request by vansisto**

### CODE REVIEW

File: src/main/java/com/intentsg/shop/services/impl/CatalogServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import com.intentsg.shop.model.Catalog;
+import com.intentsg.shop.repository.CatalogRepository;
+import com.intentsg.shop.services.CatalogService;
+import org.springframework.context.annotation.Scope;
+import org.springframework.context.annotation.ScopedProxyMode;
+import org.springframework.stereotype.Service;
+import org.springframework.web.context.WebApplicationContext;
+
+@Scope( value = WebApplicationContext.SCOPE_SESSION, proxyMode = ScopedProxyMode.INTERFACES )
```
Comment (PetroRavlinko) [inline src/main/java/com/intentsg/shop/services/impl/CatalogServiceImpl.java:11]: No reason to make its scope as the session

File: src/test/java/com/intentsg/shop/controllers/CatalogControllerIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    @MockBean
+    private CatalogService catalogService;
+    @MockBean
+    private CartService cartService;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Test
+    void whenGetCatalog_returnCatalogPage() throws Exception {
```
Comment (PetroRavlinko) [inline src/test/java/com/intentsg/shop/controllers/CatalogControllerIntegrationTest.java:29]: You don't check that it returns the catalog page. Rename test appropriately.

**Review by PetroRavlinko — CHANGES_REQUESTED** [review]

File: src/main/java/com/intentsg/shop/services/impl/CatalogServiceImpl.java (hunk 1 above)
Comment (vansisto) [reply src/main/java/com/intentsg/shop/services/impl/CatalogServiceImpl.java]: Removed

File: src/test/java/com/intentsg/shop/controllers/CatalogControllerIntegrationTest.java (hunk 2 above)
Comment (vansisto) [reply src/test/java/com/intentsg/shop/controllers/CatalogControllerIntegrationTest.java]: Renamed

File: src/main/java/com/intentsg/shop/services/impl/CatalogServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    @Override
+    public void deleteCatalog( Catalog catalogToDelete ) {
+        catalogRepository.delete( catalogToDelete );
+    }
+
+    @Override
+    public Catalog getCatalogById( Long id ) {
+        return catalogRepository.findById( id ).orElseThrow();
+    }
```
Comment (obabiy) [inline src/main/java/com/intentsg/shop/services/impl/CatalogServiceImpl.java:27]: Update Catalog method is missed

File: src/test/java/com/intentsg/shop/controllers/CatalogControllerIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    private MockMvc mockMvc;
+
+    @Test
+    void whenGetCatalog_returnStatus200() throws Exception {
+        // Given
+        given(catalogService.getCatalogById(1L)).willReturn(new Catalog());
+
+        // When
+        mockMvc.perform(get("/catalog?id=1")).andDo(print()).andExpect(status().isOk());
+    }
```
Comment (obabiy) [inline src/test/java/com/intentsg/shop/controllers/CatalogControllerIntegrationTest.java:35]: Here must be tests for all controller methods

File: src/test/java/com/intentsg/shop/services/impl/CatalogServiceTest.java [hunk 5]
```diff
@@ -14,7 +14,7 @@
 import static org.mockito.ArgumentMatchers.eq;
 import static org.mockito.BDDMockito.given;
 
-@SpringBootTest(classes = CatalogService.class)
+@SpringBootTest(classes = CatalogServiceImpl.class)
 class CatalogServiceTest {
     private static final long EXPECTED_CATALOG_ID = 1L;
```
Comment (obabiy) [inline src/test/java/com/intentsg/shop/services/impl/CatalogServiceTest.java:20]: Here must be tests for service methods

**Review by obabiy — CHANGES_REQUESTED** [review]
- Add updateCatalog function
- Cover with tests Catalog Controller and Service

**Pull request merged**